### PR TITLE
Allow replacing existing content using wagtail content import

### PR DESCRIPTION
**Pull request by kaedroho**

### CODE REVIEW

**Comment by jacobtoppm** [comment]
Hey @kaedroho, looking great! Will give this a thorough review soon, but just wanted to note that https://torchbox.github.io/wagtail-content-import/submitting_backend/ gives instructions on picker creation as part of it, so probably needs updating with this PR as well

**Review by jacobtoppm — CHANGES_REQUESTED** [review]
Looking good, just a few potential refactors + bits we should mention in the docs

File: wagtail_content_import/utils.py [hunk 1]
```diff
@@ -43,3 +43,39 @@ def create_page_from_import(request, parent_page, page_class, parsed_doc):
...
+    page.update_from_import(parsed_doc, request.user)
+
+    edit_handler = page.get_edit_handler()
+    edit_handler = edit_handler.bind_to(request=request, instance=page)
+    form_class = edit_handler.get_form_class()
+
+    next_url = get_valid_next_url_from_request(request)
+
+    form = form_class(instance=page)
+    has_unsaved_changes = False
```
[the remark this answers is not in the document]
Comment (kaedroho) [reply wagtail_content_import/utils.py:61]: It was ``False`` on create and wasn't sure why

[the remark this answers is not in the document]
Comment (kaedroho) [reply wagtail_content_import/utils.py]: Oh yes, good spot!

File: wagtail_content_import/templates/wagtail_content_import/picker_buttons_edit.html [hunk 2]
```diff
@@ -0,0 +1,33 @@
+{% if default_picker %}
```
[the remark this answers is not in the document]
Comment (kaedroho) [reply wagtail_content_import/templates/wagtail_content_import/picker_buttons_edit.html]: Updated

File: wagtail_content_import/models.py [hunk 3]
```diff
@@ -17,7 +17,12 @@ def create_from_import(cls, parsed_doc, user):
         Factory method to create the Page and populate it from a parsed document.
         """
         title = parsed_doc["title"]
-        mapper_class = cls.mapper_class
-        mapper = mapper_class()
+        mapper = cls.mapper_class()
         imported_data = mapper.map(parsed_doc["elements"], user=user)
         return cls(title=title, slug=slugify(title), body=imported_data, owner=user,)
+
+    def update_from_import(self, parsed_doc, user):
```
[the remark this answers is not in the document]
Comment (kaedroho) [reply wagtail_content_import/models.py:24]: Updated

[the remark this answers is not in the document]
Comment (jacobtoppm) [reply wagtail_content_import/models.py:24]: Thanks! What were your thoughts on:

> However, couldn't we share a fair bit between the two? Pretty much get rid of create, just create a blank instance, and just use the update_from_import instead? Feels like even if we have to make this an upgrade consideration, it should be easier to customise which fields are being imported into in one place

Ie have the creation and update methods share logic?

**Review by jacobtoppm — CHANGES_REQUESTED** [review]
Generally looking great!

- Just my remaining comment over sharing create/update logic to discuss
- A draft of how you'd like update considerations to be mentioned in release notes for custom pickers + however update logic ends up working (can be fairly quick, just for https://github.com/torchbox/wagtail-content-import/blob/master/docs/release_notes.md)

File: wagtail_content_import/models.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (kaedroho) [reply wagtail_content_import/models.py:24]: Sorry I missed this, yep I think that's a good idea. I'll have a go at implementing it!

**Comment by kaedroho** [comment]
@jacobtoppm  how is this?

Feature: The import button has been added to the page editor on existing pages, allowing editors to update them with imported content.

Upgrade consideration: Picker trigger event

If you have created a custom picker, you should update the event that it uses to trigger it to open. Previously, pickers listened for the 'click' event on the import button but we have now added a new event called 'openPicker'.

For example:

```javascript
    document.addEventListener('DOMContentLoaded', function() {
        document.querySelectorAll('[data-content-import-picker="my_picker"]').forEach(function (element) {
            // ...

            element.addEventListener('click', function() {  // <----- Change this from 'click' to 'openPicker'
                // ...
            });
        });
    });
```

See: [Submitting a new backend](https://torchbox.github.io/wagtail-content-import/submitting_backend/)

Upgrade consideration: Customised ``.create_from_import()`` method

If you have customised the ``.create_from_import()`` method on any page, you should replace this with a customisation of ``.update_from_import()`` instead. This will allow your customisation to work when the page is being updated or created.

See: [Changing Import Fields](https://torchbox.github.io/wagtail-content-import/changing_import_fields/)

**Review by jacobtoppm — APPROVED** [review]
Looks great, thanks for making those changes! Release notes sound good as well

**Pull request merged**